src/evolve.test.ts

```diff
@@ -0,0 +1,468 @@
+      count: add(1),
+      data: { elapsed: add(1), remaining: add(-1) },
+    };
+    const object = {
+      id: 10,
+      count: 10,
+      data: { elapsed: 100, remaining: 1400 },
+    };
+    const expected = {
+      id: 10,
+      count: 11,
+      data: { elapsed: 101, remaining: 1399 },
+    };
+    const result = evolve(object, transf);
+    expect(result).toEqual(expected);
+    expectTypeOf(result).toEqualTypeOf<typeof expected>();
+  });
+
+  it('does not invoke function if object does not contain the key', function () {
+    const transf = { n: add(1), m: add(1) };
+    const object = { m: 3 };
+    const expected = { m: 4 };
+    const result = evolve(object, transf);
+    expect(result).toEqual(expected);
+    expectTypeOf(result).toEqualTypeOf<typeof expected>();
+  });
+
+  it('is not destructive and is immutable', function () {
+    const transf = { n: add(1) };
+    const object = { n: 100 };
+    const expected = { n: 101 };
+    const result = evolve(object, transf);
+    expect(object).toEqual({ n: 100 });
+    expect(result).toEqual(expected);
+    expect(result).not.toBe(expected);
+    expectTypeOf(result).toEqualTypeOf<typeof expected>();
+  });
+
+  it('is recursive', function () {
+    const transf = { nested: { second: add(-1), third: add(1) } };
+    const object = { first: 1, nested: { second: 2, third: 3 } };
+    const expected = { first: 1, nested: { second: 1, third: 4 } };
+    const result = evolve(object, transf);
+    expect(result).toEqual(expected);
+    expectTypeOf(result).toEqualTypeOf<typeof expected>();
+  });
+
+  it('ignores primitive value transformations', function () {
+    const transf = { n: 2, m: 'foo' };
+    const object = { n: 0, m: 1 };
+    const expected = { n: 0, m: 1 };
+    const result = evolve(object, transf);
+    expect(result).toEqual(expected);
+    expectTypeOf(result).toEqualTypeOf<typeof expected>();
+  });
+
+  it('ignores null transformations', function () {
+    const transf = { n: null };
+    const object = { n: 0 };
+    const expected = { n: 0 };
+    const result = evolve(object, transf);
+    expect(result).toEqual(expected);
+    expectTypeOf(result).toEqualTypeOf<typeof expected>();
+  });
+
+  it('creates a new array by evolving the `array` according to the `transformation` functions', function () {
+    // NOTE:
+    // If we use tuple in `transformations` parameter,
+    // use `as const` or `evolve` can't handle typing.
+    const transf = [
+      add(-1),
+      null,
+      add(1),
+      [null, (x: string) => x + '!!'],
+    ] as const;
+    const object = [2, 2, 2, ['...', 'Go']] as const;
+    const expected = [1, 2, 3, ['...', 'Go!!']] as const;
+    let result = evolve(object, transf);
+    expect(result).toEqual(expected);
+    // FIXME: Because of limitations of working with tuple types in TypeScript.
+    expectTypeOf(result).toEqualTypeOf<{
+      0: number;
+      1: 2;
+      2: number;
+      3: { 0: '...'; 1: string };
+    }>();
```

Comment:
It might be possible to FIX it by applying `ObjectToTuple` in the [comment here](https://stackoverflow.com/a/73290716), but it would be more complicated.

Should I stop dealing with functions in arrays or keep the original Ramda implementation?